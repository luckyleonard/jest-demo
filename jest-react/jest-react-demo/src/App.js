import React from 'react';

function App() {
  return (
    <div className='app' title='prop test' data-test='app'>
      First Class
    </div>
  );
}

export default App;
