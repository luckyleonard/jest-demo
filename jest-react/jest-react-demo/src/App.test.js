import React from 'react';
import App from './App';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

test('Test <App> render a div with class name of App', () => {
  const wrapper = shallow(<App />);
  console.log(wrapper.debug());
  expect(wrapper.find('[data-test="app"]').length).toBe(1);
  expect(wrapper.find('[data-test="app"]').prop('title')).toBe('prop test');
  const container = wrapper.find('[data-test="app"]');
  expect(container).toExist();
  expect(container).toHaveProp('title', 'prop test');
});

test('Test <App> with snapshot', () => {
  const wrapper = mount(<App />);
  expect(wrapper).toMatchSnapshot();
});
