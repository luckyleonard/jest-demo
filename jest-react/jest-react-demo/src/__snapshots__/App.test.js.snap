// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <App> with snapshot 1`] = `
<App>
  <div
    className="app"
    data-test="app"
    title="prop test"
  >
    First Class
  </div>
</App>
`;
